### DISCUSSION TITLE
Miner-tools

### DISCUSSION THREAD

**Discussion by luluup777** [filecoin-project/community#352] [Mining]
Hello everyone, discuss miner tools here.

**Comment by luluup777** [comment]
Hello everyone
The most basic lotus-redo is already available, please experience it, if you have any questions, you can tell me here or raise issues.

https://github.com/luluup777/lotus-box

**Reply by luluup777** [reply]
cc sectors redo is now supported, and better tools are on the way.

**Comment by luluup777** [comment]
Hello everyone, lotus-redo tool supports multiple redo tasks in parallel, welcome to experience, I believe it is better to use. If there is any problem, please let me know

**Comment by luluup777** [comment]
Hi, everyone, I'm sorry, it has not been updated for a while, but now the lotus-wdpost tool has been tested and can be used, welcome to use, if there is any bug, I will fix it.

**Reply by luluup777** [reply]
https://github.com/luluup777/lotus-box

**Comment by luluup777** [comment]
Hi, everyone, The filecoin mining monitoring and alarm tool has been completed. Welcome to use. If there is any problem, I will follow up and fix it

**Reply by luluup777** [reply]
https://github.com/luluup777/robin

**Comment by luluup777** [comment]
Robin supports slack and lark alarm，welcome to use.

**Reply by yingzhi-zhu** [reply]
lotus-wdpost tool can support the s3？

**Reply by luluup777** [reply]
> lotus-wdpost tool can support the s3？

I am willing to support, can you provide more information, do you want to read sector data through s3?

**Reply by yingzhi-zhu** [reply]
yes,Because our sectors  is stored on S3。

**Reply by yingzhi-zhu** [reply]
thanks

**Reply by luluup777** [reply]
sorry, I have never used s3 storage sector, can you provide an s3 api example for reading sectors, so that I can support it

**Reply by yingzhi-zhu** [reply]
You can test S3 in Minio， https://docs.min.io/

**Reply by luluup777** [reply]
ok, thanks

**Comment by luluup777** [comment]
https://github.com/luluup777/wdkeep

**Reply by luluup777** [reply]
wdkeep, It is under development.
contributions welcome

**Reply by jennijuju** [reply]
Id like to learn more about this tool!

**Reply by luluup777** [reply]
> Id like to learn more about this tool!

On the way, I'm working hard to develop it, because of the Spring Festival, it's a bit delayed, I'm very sorry. please give me a little more time.

wdkeep:  Simple to quickly start the wdpost service that supports multiple miners.

I plan to do it well and I'm sure it will work well
